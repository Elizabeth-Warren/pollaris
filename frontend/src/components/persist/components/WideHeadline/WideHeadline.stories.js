import React from 'react';
import styled from 'styled-components';
import { select, text } from '@storybook/addon-knobs';
import WideHeadline, { WideHeadlineSizes } from './index';
import theme from '../../theme';

export default {
  title: 'Components/WideHeadline',
  component: WideHeadline,
  parameters: { jest: ['WideHeadline'] },
};

const sampleText = 'The quick brown fox jumps over the lazy dog.';

export const withKnobs = () => {
  const tag = text('Tag', 'h1');
  const copy = text('Text', sampleText);
  const size = select('Size', Object.values(WideHeadlineSizes), WideHeadline.defaultProps.size);
  const color = select('Color', Object.keys(theme.colors), WideHeadline.defaultProps.color);
  return (
    <div>
      <WideHeadline
        as={tag}
        size={size}
        color={color}
      >
        {copy}
      </WideHeadline>
    </div>
  );
};

export const gallery = () => {
  const copy = text('Text', sampleText);
  return (
    <div>
      {Object.entries(WideHeadlineSizes).reverse().map(([name, value], i) => (
        <WideHeadline as={`h${i + 1}`} size={value} key={name}>
          {`${name}: `}
          {copy}
        </WideHeadline>
      ))}
    </div>
  );
};

export const withStyledComponent = () => {
  const copy = text('Text', sampleText);
  const size = select('Size', Object.values(WideHeadlineSizes), WideHeadline.defaultProps.size);
  const StyledCopy = styled.div`
    color: ${theme.colors.red};
  `;
  return (
    <div>
      <WideHeadline
        as={StyledCopy}
        size={size}
      >
        {copy}
      </WideHeadline>
    </div>
  );
};
