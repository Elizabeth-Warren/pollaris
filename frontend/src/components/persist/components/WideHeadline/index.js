import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import theme from '../../theme';

export const WideHeadlineSizes = {
  SM: 'sm',
  MD: 'md',
  LG: 'lg',
};

const HeadlineWrapper = styled.div`
  color: ${({ textColor }) => theme.colors[textColor] || 'inherit'};
  font-family: ${theme.fontFamily.wide};
  font-size: ${({ size }) => {
    switch (size) {
      case WideHeadlineSizes.MD:
        return theme.fontSize.xl;
      case WideHeadlineSizes.SM:
        return theme.fontSize.sm;
      case WideHeadlineSizes.LG:
      default:
        return theme.fontSize['3xl'];
    }
  }};
  font-weight: bold;
  letter-spacing: ${({ size }) => (size === WideHeadlineSizes.SM ? '1.6px' : 0)};
  line-height: ${({ size }) => (size === WideHeadlineSizes.SM ? theme.leading.tight : theme.leading.none)};
  text-transform: uppercase;
`;

const WideHeadline = ({
  as, children, size, color, ...other
}) => (
  <HeadlineWrapper
    as={as}
    size={size}
    textColor={color}
    {...other}
  >
    {children}
  </HeadlineWrapper>
);

WideHeadline.propTypes = {
  /** Child text or elements */
  children: PropTypes.node.isRequired,
  /** HTML tag name or React element. */
  as: PropTypes.oneOfType([PropTypes.string, PropTypes.element]).isRequired,
  /** Size of the headline */
  size: PropTypes.oneOf(Object.values(WideHeadlineSizes)),
  /** Color of the headline */
  color: PropTypes.oneOf(['inherit', ...Object.keys(theme.colors)]),
};

WideHeadline.defaultProps = {
  size: WideHeadlineSizes.LG,
  color: 'inherit',
};

export default WideHeadline;
