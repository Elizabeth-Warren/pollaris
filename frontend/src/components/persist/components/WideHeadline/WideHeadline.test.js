import React from 'react';
import renderer from 'react-test-renderer';
import WideHeadline, { WideHeadlineSizes } from './index';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <div>
        {Object.entries(WideHeadlineSizes).reverse().map(([name, value]) => (
          <WideHeadline as="h1" size={value} key={name}>
            {`${name}: `}
            Hello world
          </WideHeadline>
        ))}
      </div>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
