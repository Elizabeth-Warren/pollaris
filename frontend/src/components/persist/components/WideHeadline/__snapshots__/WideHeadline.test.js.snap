// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.c0 {
  color: inherit;
  font-family: 'Ringside Extra Wide SSm A','Ringside Extra Wide SSm B','Helvetica',sans-serif;
  font-size: 40px;
  font-weight: bold;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  line-height: 1;
  text-transform: uppercase;
}

.c1 {
  color: inherit;
  font-family: 'Ringside Extra Wide SSm A','Ringside Extra Wide SSm B','Helvetica',sans-serif;
  font-size: 26px;
  font-weight: bold;
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  line-height: 1;
  text-transform: uppercase;
}

.c2 {
  color: inherit;
  font-family: 'Ringside Extra Wide SSm A','Ringside Extra Wide SSm B','Helvetica',sans-serif;
  font-size: 14px;
  font-weight: bold;
  -webkit-letter-spacing: 1.6px;
  -moz-letter-spacing: 1.6px;
  -ms-letter-spacing: 1.6px;
  letter-spacing: 1.6px;
  line-height: 1.17;
  text-transform: uppercase;
}

<div>
  <h1
    className="c0"
    size="lg"
  >
    LG: 
    Hello world
  </h1>
  <h1
    className="c1"
    size="md"
  >
    MD: 
    Hello world
  </h1>
  <h1
    className="c2"
    size="sm"
  >
    SM: 
    Hello world
  </h1>
</div>
`;
